### PR TITLE
samples: Bluetooth: df: Add info about max ant sw pattern len Kconfig

### PR DESCRIPTION
**Pull request by ppryga**
There is a maximum length of antenna switch pattern defined by
CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN. It limits number of
antenna ids that may be used by an application.

There were no information about it in direction finding samples
README.rst files.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

### CODE REVIEW

File: samples/bluetooth/direction_finding_central/README.rst [hunk 1]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
 
 If the number of switch-sample periods is greater than the number of stored switching patterns, then the radio loops back to the first pattern.
 
+Maximum length of antenna switch pattern is limited by :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
```
Comment (alwa-nordic) [inline samples/bluetooth/direction_finding_central/README.rst:88]: The form "The maximum .. is limited by .." is redundant. I suggest removing "maximum":
```suggestion
The length of the antenna switch pattern is limited by :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
```

File: samples/bluetooth/direction_finding_central/README.rst [hunk 2]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
...
+If required length of the antenna switching pattern is greater than default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` add the Kconfig to a board configuration file.
+For example for :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is required antanna switch pattern length, to :file:`nrf52833dk_nrf52833.conf`
```
Comment (alwa-nordic) [inline samples/bluetooth/direction_finding_central/README.rst:90]: We are missing articles in this sentence.
```suggestion
If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` add the Kconfig to a board configuration file.
For example for :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switch pattern length, to :file:`nrf52833dk_nrf52833.conf`
```

I also have an alternative formulation below that I think is simpler. (If you prefer.)

```suggestion
The value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` can be set in a board configuration file.
E.g. for :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` with the desired value to :file:`nrf52833dk_nrf52833.conf`
```

**Review by alwa-nordic — CHANGES_REQUESTED** [review]

File: samples/bluetooth/direction_finding_central/README.rst (hunk 1 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/README.rst]: Corrected in all README.rst

File: samples/bluetooth/direction_finding_central/README.rst (hunk 2 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/README.rst]: Corrected in all README.rst

**Comment by ppryga** [comment]
@KAGA164 @mia-ko could you take a look?

File: samples/bluetooth/direction_finding_central/README.rst [hunk 3]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
 
 If the number of switch-sample periods is greater than the number of stored switching patterns, then the radio loops back to the first pattern.
 
+The length of an antenna switch pattern is limited by :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
```
Comment (KAGA164) [inline samples/bluetooth/direction_finding_central/README.rst:88]:
```suggestion
The length of an antenna switch pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
```

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst [hunk 4]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
 
 If the number of switch-sample periods is greater than the number of stored switching patterns, then the radio loops back to the first pattern.
 
+The length of an antenna switch pattern is limited by :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
```
Comment (KAGA164) [inline samples/bluetooth/direction_finding_connectionless_rx/README.rst:88]:
```suggestion
The length of an antenna switch pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
```

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst [hunk 5]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
...
+If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` add the Kconfig to a board configuration file.
```
Comment (KAGA164) [inline samples/bluetooth/direction_finding_connectionless_rx/README.rst:89]:
```suggestion
If the required length of the antenna switching pattern is greater than the default value of the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set this option to required value in a board configuration file.
```

Comment (KAGA164) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst:89]: "The Kconfig" in this context is a little confusing to me

Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst]: I've changed it to:
```
If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` with required value in a board configuration file.
```

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst (hunk 4 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/README.rst]: Done

File: samples/bluetooth/direction_finding_central/README.rst (hunk 3 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/README.rst]: Done

File: samples/bluetooth/direction_finding_central/README.rst [hunk 6]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
 
 If the number of switch-sample periods is greater than the number of stored switching patterns, then the radio loops back to the first pattern.
 
+The length of an antenna switch pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
+If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` with required value in a board configuration file.
+For example for :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switch pattern length, to :file:`nrf52833dk_nrf52833.conf`.
+
```
Comment (mia-ko) [inline samples/bluetooth/direction_finding_central/README.rst:91]:
```suggestion
The length of the antenna switching pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option.
If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option to the required value in the board configuration file.
For example, for the :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switching pattern length, to the :file:`nrf52833dk_nrf52833.conf` file.

```

File: samples/bluetooth/direction_finding_connectionless_rx/README.rst [hunk 7]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
 
 If the number of switch-sample periods is greater than the number of stored switching patterns, then the radio loops back to the first pattern.
 
+The length of an antenna switch pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
+If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` with required value in a board configuration file.
+For example for :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switch pattern length, to :file:`nrf52833dk_nrf52833.conf`.
+
```
Comment (mia-ko) [inline samples/bluetooth/direction_finding_connectionless_rx/README.rst:91]:
```suggestion
The length of the antenna switching pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option.
If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option to the required value in the board configuration file.
For example, for the :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switching pattern length, to the :file:`nrf52833dk_nrf52833.conf` file.

```

File: samples/bluetooth/direction_finding_connectionless_tx/README.rst [hunk 8]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
 
 If the number of switch-sample periods is greater than the number of stored switching patterns, then the radio loops back to the first pattern.
 
+The length of an antenna switch pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
+If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` with required value in a board configuration file.
+For example for :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switch pattern length, to :file:`nrf52833dk_nrf52833.conf`.
+
```
Comment (mia-ko) [inline samples/bluetooth/direction_finding_connectionless_tx/README.rst:91]:
```suggestion
The length of the antenna switching pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option.
If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option to the required value in the board configuration file.
For example, for the :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switching pattern length, to the :file:`nrf52833dk_nrf52833.conf` file.

```

File: samples/bluetooth/direction_finding_peripheral/README.rst [hunk 9]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
 
 If the number of switch-sample periods is greater than the number of stored switching patterns, then the radio loops back to the first pattern.
 
+The length of an antenna switch pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`.
+If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` with required value in a board configuration file.
+For example for :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switch pattern length, to :file:`nrf52833dk_nrf52833.conf`.
+
```
Comment (mia-ko) [inline samples/bluetooth/direction_finding_peripheral/README.rst:91]:
```suggestion
The length of the antenna switching pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option.
If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option to the required value in the board configuration file.
For example, for the :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switching pattern length, to the :file:`nrf52833dk_nrf52833.conf` file.

```

**Review by mia-ko — CHANGES_REQUESTED** [review]
Sorry for such a delay, due to sick leave :(

**Comment by ppryga** [comment]
@mia-ko No worries. I'm glad you are better now. I've applied your suggestions and made you a co-author of the commit.

File: samples/bluetooth/direction_finding_central/README.rst [hunk 10]
```diff
@@ -85,6 +85,10 @@ This also means that, for example, when using four GPIOs, the pattern count cann
 
 If the number of switch-sample periods is greater than the number of stored switching patterns, then the radio loops back to the first pattern.
 
+The length of the antenna switching pattern is limited by the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option.
+If the required length of the antenna switching pattern is greater than the default value of :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`, set the :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN` option to the required value in the board configuration file.
+For example, for the :ref:`nRF52833 DK <ug_nrf52>` add :kconfig:`CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN`=N, where N is the required antenna switching pattern length, to the :file:`nrf52833dk_nrf52833.conf` file.
```
[the remark this answers is not in the document]
Comment (mia-ko) [reply samples/bluetooth/direction_finding_central/README.rst:90]: We have to introduce space before =N
Without it, the text is rendered in a wrong way it seems.
This should be adjusted in all four files :)

[the remark this answers is not in the document]
Comment (mia-ko) [reply samples/bluetooth/direction_finding_central/README.rst]: I pushed a commit for this that you can squash :)

[the remark this answers is not in the document]
Comment (ppryga) [reply samples/bluetooth/direction_finding_central/README.rst]: Thank you, commit is squashed.

**Comment by ppryga** [comment]
I've added another commit to piggy-back the PR. The commit fixes error in example evaluation of total number of collected samples.

**Comment by ppryga** [comment]
@alwa-nordic @mia-ko @KAGA164 have a look on the PR please. I'd like to have it closed since it is hanging here for a while.

**Review by KAGA164 — APPROVED** [review]
Looks good

**Review by mia-ko — APPROVED** [review]
The added content looks good.
We need entries in the release-notes-changelog.rst file?

**Comment by ppryga** [comment]
@mia-ko 
> We need entries in the release-notes-changelog.rst file?

Do we add information in release-notes-changelog.rst if documentation is updated or errors there are fixed?

**Comment by ppryga** [comment]
@alwa-nordic could you re-review?

**Review by alwa-nordic — APPROVED** [review]

**Pull request merged**